account_analytic_distribution_manual/models/base.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2024 Tecnativa - Carlos Lopez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import json
+
+from lxml import etree
+
+from odoo import api, models
+from odoo.tools.misc import frozendict
+
+
+class AnalyticMixin(models.AbstractModel):
+    _inherit = "base"
+
+    @api.model
+    def get_view(self, view_id=None, view_type="form", **options):
+        def add_field(node, view_type, res_model):
+            attribute = "column_invisible" if view_type == "tree" else "invisible"
+            modifiers = json.dumps({attribute: True})
+            field_options = {
+                "name": manual_distribution_field_name,
+                "modifiers": modifiers,
+            }
+            field_element = etree.SubElement(node, "field", field_options)
+            new_arch, new_models = View.postprocess_and_fields(field_element, res_model)
+            _merge_view_fields(all_models, new_models)
+            return field_element
+
+        def model_has_field(model):
+            return manual_distribution_field_name in self.env[model]._fields
+
+        def _merge_view_fields(all_models, new_models):
+            """Merge new_models into all_models. Both are {modelname(str) ➔ fields(tuple)}."""
+            for model, view_fields in new_models.items():
+                if model in all_models:
+                    all_models[model] = tuple(set(all_models[model]) | set(view_fields))
+                else:
+                    all_models[model] = tuple(view_fields)
+
+        result = super().get_view(view_id=view_id, view_type=view_type, **options)
+        if view_type in ["tree", "form"]:
+            View = self.env["ir.ui.view"]
+            manual_distribution_field_name = "manual_distribution_id"
+            all_models = result["models"].copy()  # {modelname(str) ➔ fields(tuple)}
```

Comment:
Please restrict this processing the maximum and optimize it. We are injecting this code for every view...

Reply:
I don't understand what optimization I can implement. I need to check if the current model inherits from the mixin or if this model has a one2many field that inherits from the mixin. I don't see how this code can be more optimized. Can you please give an example?

Reply:
For example, don't copy this, which is a memory drainer and not sure why do you need to do it.

Reply:
This variable `result["models"]` is a `frozendict`, so I can't modify it without making a new copy.